### ISSUE TITLE
Slight delay with certain SAPI5 voices in NVDA 2025.3 beta, even when rate boost is disabled

### ISSUE THREAD

**Issue by HBM301** [nvaccess/nvda#18811]
Since updating to NVDA 2025.3 beta, a slight but consistent delay has been observed when using certain SAPI5 voices.
This delay occurs even when the "rate boost" feature is turned off, indicating that the issue is not directly related to that setting.
In NVDA 2025.2, speech output with the affected voices was immediate. In 2025.3 beta, however, there is a slight but noticeable delay before speech begins.
### Steps to reproduce:
- Switch to a SAPI5 voice known to be affected (e.g.
[Sao Mai Voice, which can be downloaded and installed from the official website](https://saomaicenter.org/sites/default/files/inline-files/SaoMai_voice1.0.exe)
- Ensure that "rate boost" is disabled
- Perform actions that trigger speech output (e.g., navigate text, execute commands, or use continuous reading).
- Observe a slight delay before speech starts.
### Actual behavior:

Speech output begins with a slight delay, even when rate boost is disabled
### Expected behavior:

Speech output should begin immediately, as it did in NVDA 2025.2, with no additional delay when rate boost is off
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2025.3 beta2
#### Windows version:
Windows 11 24H2 (AMD64) build 26100.4946
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, 2025.2 and 2025.1
#### If NVDA add-ons are disabled, is your problem still occurring?
Don't install any addon
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
not tested
### Additional Notes:
- The issue was first noticed after upgrading to NVDA 2025.3 beta
- So far, the delay has been observed specifically with Sao Mai Voice, while other SAPI5 voices appear unaffected
- The silence trimming feature, introduced in NVDA 2025.1, is still effective in 2025.3 beta. However, overall speech responsiveness feels slower compared to  2025.1 and 2025.2

**Comment by gexgd0419** [comment]
I tried Sao Mai Voice, and I noticed that those voices cannot speak the keyboard keys during typing. This is because that those voices don't support "spelling each character" (or the `CharacterModeCommand`), and instead of ignoring the `<spell>` tag and speaking the text inside it, those voices ignore the entire `<spell>` tag with the content, so nothing will be spoken when  `CharacterModeCommand` is used. Fortunately there's a workaround: turning off "Use spelling functionality if supported" (Sử dụng chức năng đánh vần nếu có hỗ trợ) in the speech settings will make this work.

Back to the topic. Is [this snapshot version](https://github.com/nvaccess/nvda/actions/runs/17232252854/artifacts/3851305732) from #18815 better with rate boost disabled?

**Comment by HBM2001** [comment]
@gexgd0419 I have completely disabled the spelling functionality for this voice, even when using previous NVDA versions. You might want to try an earlier version, such as 2025.2 to observe the difference in responsiveness. Personally, I also rarely use the rate boost feature

**Comment by HBM2001** [comment]
I have also tested
[This snapshot](https://github.com/nvaccess/nvda/actions/runs/17232252854/artifacts/3851305732), but I haven't noticed any difference.

**Comment by gexgd0419** [comment]
Could you please test the following alpha versions and check whether each version has the extra delay or not? (They might have other bugs, but we only focus on the delay here.)

- [Alpha 37633 at commit 4bd5a3dd](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-37633,4bd5a3dd.exe)
- [Alpha 37658 at commit 4e60eebe](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-37658,4e60eebe.exe)
- [Alpha 51845 at commit 8dffe620](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-51845,8dffe620.exe)
- [Alpha 52140 at commit 9eabf487](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-52140,9eabf487.exe)

**Comment by HBM2001** [comment]
@gexgd0419 I have created portable copies for all 4 alpha versions, and from what I can see:
- [Alpha-37633,4bd5a3dd](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-37633,4bd5a3dd.exe)
- [Alpha-37658,4e60eebe](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-37658,4e60eebe.exe)

Both versions, I no longer notice any delay. The speech responsiveness now works exactly as I expected

And with the other two versions, the delay is still very noticeable, I tested all versions with rate boost disabled.

**Comment by gexgd0419** [comment]
What about [this snapshot build](https://github.com/nvaccess/nvda/actions/runs/17242268224/artifacts/3854821080)?

**Comment by HBM2001** [comment]
@gexgd0419 Hi, I tested this
[snapshot build](https://github.com/nvaccess/nvda/actions/runs/17242268224/artifacts/3854821080)and the delay has clearly improved a bit, It would be great if the delay could be improved as in the alpha versions above

**Comment by HBM2001** [comment]
HI
@gexgd0419 apologies for pinging you on this issue again
With the release of NVDA 2026.1 beta, I gave it a try and discovered an interesting bug regarding the responsiveness of the SAPI5 voice I am using (I also verified this on NVDA 2025.3.2).
When I switch NVDA's interface language to Vietnamese, speech responsiveness works exactly as expected. However, when I switch the interface back to English, a delay is clearly noticeable.
Would you have some time to investigate this specific behavior again?
thanks!

**Comment by seanbudd** [comment]
is this a 32 bit or 64 bit synthesizer?

**Comment by HBM2001** [comment]
Hi 
@seanbudd This is a 32 bit Synthesizer, it's quite an old speech synthesizer

**Comment by seanbudd** [comment]
@michaelDCurran - this might be a new issue with the new 32bit synth infrastructure
